### ISSUE TITLE
bug: Examples to customize the lualine config are not working

### ISSUE THREAD

**Issue by HendrikThePendric** [LazyVim/LazyVim#4544]
### Did you check docs and existing issues?

- [x] I have read all the LazyVim docs
- [x] I have updated the plugin to the latest version before submitting this issue
- [x] I have searched the existing issues of LazyVim
- [x] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

v0.10.2

### Operating system/version

Ubuntu 24.04.1 LTS

### Describe the bug

As also noted in [this SO post](https://stackoverflow.com/a/78727919), the examples to customize the lualine config are not working. I am new to LazyVim and its ecosystem, but customization of other plugins did work exactly as described in the docs/examples.

### Steps To Reproduce

1. Create a lua file in the `lua/plugins` dir with the same content as one of the examples, i.e.
```lua
return {
  "nvim-lualine/lualine.nvim",
  event = "VeryLazy",
  opts = function(_, opts)
    table.insert(opts.sections.lualine_x, "TEST")
  end,
}
```

2. Open a file, so that `lualine_x` becomes visible (according to the [the lualine docs](https://github.com/nvim-lualine/lualine.nvim?tab=readme-ov-file#default-configuration), `lualine_x` is the part that shows file encoding etc.)
3. Observe that the expected text "TEST" is not present in section that shows file encoding.

### Expected Behavior

I would have expected to see the "TEST" string in the lualine

**Comment by dpetka2001** [comment]
You really need to read the Lualine [docs](https://github.com/nvim-lualine/lualine.nvim?tab=readme-ov-file#lua-functions-as-lualine-component) more carefully. What you describe is not a valid Lualine component. LazyVim docs are meant as a reference for how to configure LazyVim. You have to use them together with the docs of the plugin you're interested in, which describe how the plugin works in more detail.

The following should work
```lua
return {
  {
    "nvim-lualine/lualine.nvim",
    opts = function(_, opts)
      table.insert(opts.sections.lualine_x, {
        function()
          return "test"
        end,
      })
    end,
  },
}
```

**Comment by HendrikThePendric** [comment]
Fair comment about  having to read the lualine docs better. The docs you link to indeed contain relevant info, I managed to get things to work as I intended now. 

Having said that, I personally would have expected to be able to copy-paste these examples and that they would work, especially given the fact that the file structure kind of suggests that "this is real code": commenting out line 3 would actually apply the full example config.

Not a big deal though. I think this is a great project and appreciate all the effort of maintaining it.
